File: forum_back/src/main/kotlin/com/example/forum_back/Forum.java
package com.example.forum_back;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@Table(name = "forum")
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Forum {

    @Id
    private Long id;
    private String title;
    private String description;
    private String author;
    private String category;

    public static Forum CreateNewForum(String title, String description, String author, String category) {
        Forum forum = new Forum();
        forum.title = title;
        forum.description = description;
        forum.author = author;
        forum.category = category;
        return forum;
    }

}

File: forum_back/src/main/kotlin/com/example/forum_back/entity/Forum.kt
package com.example.forum_back.entity

import com.example.forum_back.dto.ForumUpdateRequestDto
import jakarta.persistence.*
import jakarta.persistence.GenerationType.*

@Entity
@Table(name = "forum")
class Forum constructor(
    var title: String,
    var description: String,
    var author: String,
    @Id
    @GeneratedValue(strategy = IDENTITY)
    val id: Long? = null
) {

    init {
        if (title.isBlank()) {
            throw IllegalArgumentException("title cannot be blank")
        }
    }

    fun updateForum(forumUpdateRequestDto: ForumUpdateRequestDto){
        this.title = forumUpdateRequestDto.title;
        this.description = forumUpdateRequestDto.description;
    }

}